src/index.web.tsx

```diff
@@ -0,0 +1,122 @@
+type Profiler = any;
+declare global {
+  interface Window {
+    Profiler: Profiler;
+  }
+}
+
+const NOT_STARTED_PROFILER = {
+  stop: (): Promise<ProfilingTrace> => {
+    console.warn('The profiler has not been started');
+
+    return Promise.resolve({
+      frames: [],
+      resources: [],
+      samples: [],
+      stacks: [],
+    });
+  },
+};
+let profiler = NOT_STARTED_PROFILER;
+
+const SAMPLING_INTERVAL = 10;
+const MAX_BUFFER_SIZE = 10000;
+const MICRO_SECONDS_IN_MILLISECONDS = 1000;
+
+export function startProfiling(): boolean {
+  if (typeof window.Profiler !== 'function') {
+    console.warn(
+      'Profiler is not available in this browser. Ignoring this profiler run...'
+    );
+
+    return false;
+  }
+
+  profiler = new Profiler({
+    sampleInterval: SAMPLING_INTERVAL,
+    maxBufferSize: MAX_BUFFER_SIZE,
+  });
+
+  return true;
+}
+
+export async function stopProfiling(saveToDownloads = false): Promise<string> {
+  const trace = await profiler.stop();
+
+  profiler = NOT_STARTED_PROFILER;
+
+  if (saveToDownloads) {
+    downloadJsonFile(
+      convertToHermesProfilerFormat(trace),
+      `trace-${new Date().toISOString()}`
+    );
+  }
+
+  return '';
+}
+
+type Sample = {
+  stackId?: number;
+  timestamp: number;
+};
+type Stack = {
+  frameId: number;
+  parentId?: number;
+};
+type Frame = {
+  name: string;
+  line?: number;
+  column?: number;
+  resourceId?: number;
+};
+type ProfilingTrace = {
+  frames: Frame[];
+  resources: string[];
+  samples: Sample[];
+  stacks: Stack[];
+};
+
+function downloadJsonFile(
+  exportObj: Record<string, unknown>,
+  exportName: string
+) {
+  const dataStr =
+    'data:text/json;charset=utf-8,' +
+    encodeURIComponent(JSON.stringify(exportObj));
+  const downloadAnchorNode = document.createElement('a');
+
+  downloadAnchorNode.setAttribute('href', dataStr);
+  downloadAnchorNode.setAttribute('download', exportName + '.cpuprofile.txt');
+  document.body.appendChild(downloadAnchorNode); // required for Firefox
+  downloadAnchorNode.click();
+  downloadAnchorNode.remove();
+}
+
+function convertToHermesProfilerFormat(profilingTrace: ProfilingTrace) {
+  const { frames, resources, samples, stacks } = profilingTrace;
+
+  return {
+    traceEvents: [],
+    samples: samples.map((sample) => ({
+      cpu: '-1',
+      name: '',
+      ts: sample.timestamp * MICRO_SECONDS_IN_MILLISECONDS,
+      pid: 1,
+      tid: '1',
+      weight: '1',
```

Comment:
is that really working correctly with the weight being set to '1'?

Reply:
@hannojg yes, well at least I think so and let me explain why I think so :)

1️⃣  I analysed Hermes stack trace and I didn't find a case when `weight !== '1'`.
2️⃣ From a report that we receive there is no way to extract `weight` just because such value doesn't exist in the original data
3️⃣ Since we measure only time based metrics, I don't think that one time metric can be more important/to have bigger weight than the other?